### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: HappyTravel.RequestAuditLog/Infrastructure/VaultClientFactory.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                BaseUrl = new Uri(configuration[configuration["Vault:Endpoint"]]),
+                Engine = configuration["Vault:Engine"],
+                Role = configuration["Vault:Role"]
+            };
+
+            return new VaultClient.VaultClient(vaultOptions);
+        }
+
+
+        public static void Login(this IVaultClient client, IConfiguration configuration)
```
Comment (mingazhev) [inline HappyTravel.RequestAuditLog/Infrastructure/VaultClientFactory.cs:22]: Factory typically only creates something. Maybe rename class to something like Helper

File: HappyTravel.RequestAuditLog/Models/AuditLogEntry.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+using System;
+
+namespace HappyTravel.RequestAuditLog.Models
+{
+    public record AuditLogEntry
```
Comment (mingazhev) [inline HappyTravel.RequestAuditLog/Models/AuditLogEntry.cs:5]: HttpRequestAuditLogEntry?

File: HappyTravel.RequestAuditLog/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+            app.UseHttpsRedirection();
+
+            app.UseRouting();
+
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapHealthChecks("/health");
```
Comment (mingazhev) [inline HappyTravel.RequestAuditLog/Startup.cs:49]: do healthchecks need authorization?

**Review by mingazhev — APPROVED** [review]
Mostly approved, need @kirillta opinion on using records instead of readonly structs

File: HappyTravel.RequestAuditLog/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  "AllowedHosts": "*",
+  "Vault": {
+    "Endpoint": "HTDC_VAULT_ENDPOINT",
+    "Engine": "secrets",
+    "Role": "fukuoka",
+    "Token": "HTDC_VAULT_TOKEN"
+  },
+  "Elasticsearch": {
+    "Url": "fukuoka/elasticsearch"
```
Comment (kirillta) [inline HappyTravel.RequestAuditLog/appsettings.json:17]: We use `Endpoint` across the services

**Review by kirillta — CHANGES_REQUESTED** [review]

File: HappyTravel.RequestAuditLog/Startup.cs (hunk 3 above)
Comment (DFSko) [reply HappyTravel.RequestAuditLog/Startup.cs:49]: What for? If service can not be accessible from internet it not be needed

File: HappyTravel.RequestAuditLog/Infrastructure/VaultClientFactory.cs (hunk 1 above)
Comment (DFSko) [reply HappyTravel.RequestAuditLog/Infrastructure/VaultClientFactory.cs]: Fixed

File: HappyTravel.RequestAuditLog/Models/AuditLogEntry.cs (hunk 2 above)
Comment (DFSko) [reply HappyTravel.RequestAuditLog/Models/AuditLogEntry.cs]: Fixed

File: HappyTravel.RequestAuditLog/Controllers/AuditLogsController.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    public class AuditLogsController : ControllerBase
+    {
+        public AuditLogsController(IAuditLogService auditLogService)
+        {
+            _auditLogService = auditLogService;
+        }
+
+
+        [HttpPost("requests")]
+        public async Task<ActionResult> AddEntry(LogEntryRequest request, CancellationToken token)
```
[the remark this answers is not in the document]
Comment (DFSko) [reply HappyTravel.RequestAuditLog/Controllers/AuditLogsController.cs]: Fixed

File: HappyTravel.RequestAuditLog/Models/AuditLogEntry.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            DateTime = dateTime;
+        }
+
+
+        public string? Url { get; }
+        public string? RequestHeaders { get; }
+        public string? ResponseHeaders { get; }
+        public string? RequestBody { get; }
+        public string? ResponseBody { get; }
+        public int? StatusCode { get; }
```
[the remark this answers is not in the document]
Comment (DFSko) [reply HappyTravel.RequestAuditLog/Models/AuditLogEntry.cs]: Fixed

File: HappyTravel.RequestAuditLog/appsettings.json (hunk 4 above)
Comment (DFSko) [reply HappyTravel.RequestAuditLog/appsettings.json]: Fixed

**Review by mingazhev — APPROVED** [review]
With small comments

File: HappyTravel.RequestAuditLog/Startup.cs (hunk 3 above)
Comment (mingazhev) [reply HappyTravel.RequestAuditLog/Startup.cs:49]: Health checks are mostly needed not for internet but for internal infrastructure dashboards

**Review by kirillta — APPROVED** [review]

File: HappyTravel.RequestAuditLog/Infrastructure/ElasticSearchExtension.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+using System;
+using HappyTravel.VaultClient;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Nest;
+
+namespace HappyTravel.RequestAuditLog.Infrastructure
+{
+    public static class ElasticSearchExtension
```
[the remark this answers is not in the document]
Comment (DFSko) [reply HappyTravel.RequestAuditLog/Infrastructure/ElasticSearchExtension.cs]: Fixed

File: HappyTravel.RequestAuditLog/Models/HttpRequestAuditLogEntry.cs [hunk 8]
```diff
@@ -0,0 +1,15 @@
+using System.Collections.Generic;
+
+namespace HappyTravel.RequestAuditLog.Models
+{
+    public record HttpRequestAuditLogEntry
+    {
+        public string Url { get; init; }
```
[the remark this answers is not in the document]
Comment (DFSko) [reply HappyTravel.RequestAuditLog/Models/HttpRequestAuditLogEntry.cs:10]: Fixed

**Pull request merged**